enzyme/Enzyme/CApi.cpp

```diff
@@ -340,6 +340,21 @@ void EnzymeRegisterAllocationHandler(char *Name, CustomShadowAlloc AHandle,
     };
 }
 
+
+/// This is the main entry point to register a custom derivative for language frontends.
+/// It should be prefered over trying to register custom-derivatives in the llvm-ir module.
```

Comment:
I wouldn't say necessarily that it's preferred, but it's certainly much more general. It supports arbitrary LLVM code injection, with full access to all Enzyme and LLVM analyses (such as activity analysis, differential use analysis, alias analysis, etc)